src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DSDropDownBase.cs

```diff
@@ -74,7 +73,9 @@ public void Dispose()
         /// <param name="e"></param>
         void combo_DropDownOpened(object sender, EventArgs e)
         {
+            string itemName = model.getNameOfSelectedDropDownItem();
             this.model.PopulateItems();
+            model.SelectedIndex = model.getIndexOfItem(itemName); 
```

Comment:
Can we not just do this?

```csharp
var currentIndex = model.SelectedIndex;
model.PopulateItems(currentIndex);
```

Then in the new `PopulateItems` method, try to select the item if the given `currentIndex` is greater or equal to zero, and less than the number of total items:

```csharp
void PopulateItems(int currentIndex)
{
    if (currentIndex >= 0 || (currentIndex < itemCount))
    {
        // reselect the item if it exists...
    }
}
```